llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1330,6 +1330,13 @@ class VPInstruction : public VPRecipeWithIRFlags {
     assert(Opcode == Instruction::Or && "only OR opcodes can be disjoint");
   }
 
+  VPInstruction(VPValue *Ptr, VPValue *Offset, bool InBounds, DebugLoc DL = {},
+                const Twine &Name = "")
+      : VPRecipeWithIRFlags(VPDef::VPInstructionSC,
+                            ArrayRef<VPValue *>({Ptr, Offset}),
```

Comment:
I'm not sure about this ArrayRef of a temporary: isn't ArrayRef non-owning, and wouldn't this temporary `{Ptr, Offset}` get destroyed?